### ISSUE TITLE
Universal with ngZone noop

### ISSUE THREAD

**Issue by moczix** [angular/universal#1856]
Hello. on my browser app i use ngZone noop in boostraping appModule
How can i achieve the same for universal ssr? i didnt see that option at CommonEngine

**Comment by alan-agius4** [comment]
Hi @moczix, this is not possible. I'll add this a feature request.

**Comment by moczix** [comment]
well, i made some custom changes in CommonEngine and have now universal without zone :dancers: 
ill share it later as plugin, here

**Comment by sod** [comment]
It's possible. This is our render function:

```typescript
async function renderModuleWithoutZone(app: Type<unknown>, platformRef: PlatformRef): Promise<string> {
    const moduleRef = await platformRef.bootstrapModule(app, {ngZone: 'noop'});

    // leapfrog the Promises scheduled by @angular/router
    await new Promise((resolve) => setImmediate(resolve));

    const platformState = platformRef.injector.get(PlatformState);

    const callbacks = moduleRef.injector.get(BEFORE_APP_SERIALIZED, undefined);
    if (callbacks) {
        await Promise.all(
            callbacks
                .map((callback) => callback())
                .filter((result?: any): result is Promise<unknown> => typeof result?.then === 'function'),
        );
    }

    const result = platformState.renderToString();
    moduleRef.destroy();

    return result;
}
```

And zone.js catches uncaught errors by default instead of crashing the server. So if you are used to that, you should do that too

```typescript
process.on('uncaughtException', function (error) {
    yourLoggerCallbackOfChoice(error);
});
```

**Comment by moczix** [comment]
@sod this is bad. I mean it could work, but it is not the angular way its look more like hack

**Comment by moczix** [comment]
i forgot to post my way to solve this, so this is the way:
https://github.com/moczix/angular-universal-zoneless

Angular team: if you feel that this is good, you can include this in future universal release, just let me know

**Comment by BenLune** [comment]
Hi @moczix ,
Thanks a lot for your help. Unfortunatly, your version seems to be quite too old, and it doesn't work with the latest version of Angular (14). I try to look for another solution.

I hope we'll get soon a way to manage this ZoneLess SSR in a more official way.

**Comment by BenLune** [comment]
Hi,
I finally succeeded in making my app working zoneless with Universal. I added a listener to route navigation event to trigger ChangeDetection, greatly inspired by this [example](https://github.com/tastejs/angular-movies/blob/main/projects/movies/src/app/shared/zone-less/zone-less-routing.service.ts) project powered by rx-angular team.
I spent time understanding that Universal waits for the route being built to consider the app rendering finished.
So, in SSR mode, I created an SSRResolver, triggering requests in order to get all my data when the route is rendered.
But in no SSR mode (my app is a PWA, so it can boot with nothing from SSR), I want to trigger my data requests when the route in decided, not waiting for the data.
'Hope it helps someone...
